nullaway/src/main/java/com/uber/nullaway/handlers/AssertionHandler.java

```diff
@@ -97,6 +132,50 @@ private boolean isMethodAssertThat(Symbol.MethodSymbol methodSymbol) {
     return matchesMethod(methodSymbol, assertThat, assertThatOwner);
   }
 
+  private boolean isMethodHamcrestAssertThat(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, assertThat, hamcrestAssertClass);
+  }
+
+  private boolean isMethodJunitAssertThat(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, assertThat, junitAssertClass);
+  }
+
+  private boolean isMatcherIsNotNull(Node node) {
+    // Matches with
+    //   * is(not(nullValue()))
+    //   * is(notNullValue())
+    if (matchesMatcherMethod(node, isMatcher)) {
+      // All overloads of `is` method have exactly one argument.
+      return isMatcherNotNull(((MethodInvocationNode) node).getArgument(0));
+    }
+    return false;
+  }
+
+  private boolean isMatcherNotNull(Node node) {
+    // Matches with
+    //   * not(nullValue())
+    //   * notNullValue()
+    if (matchesMatcherMethod(node, notMatcher)) {
+      // All overloads of `not` method have exactly one argument.
+      return isMatcherNull(((MethodInvocationNode) node).getArgument(0));
+    }
+    return matchesMatcherMethod(node, notNullValueMatcher);
+  }
+
+  private boolean isMatcherNull(Node node) {
+    // Matches with nullValue()
+    return matchesMatcherMethod(node, nullValueMatcher);
```

Comment:
Does this also need a case for `not(...)`. Is `assertThat(o, is(not(not(notNullValue()))))` valid code? Do we care?

Reply:
It should be pretty bad code in that case. If ever that comes up, I think we could safely ask the users to remove the double negation here for it to work.

If for some weird reason this becomes a necessity, it wouldn't be too hard to add this. I just don't want to add it without knowing if there is a real reason to write code like this.

Reply:
Fair enough. I agree :)